src/Microsoft.DotNet.Build.CloudTestTasks/FilterForUpload.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.DotNet.Build.CloudTestTasks
+{
+    public sealed class FilterForUpload : Task
+    {
+        [Required]
+        public string TestOutputDirListsLocation { get; set; }
+
+
+        [Required]
+        public string FilterToTestTFM { get; set; }
```

Comment:
@MattGal This is primarily to resolve the issue with the current corefx runs. Today we build all the tests that have more than one entry in the builds file more than once, each with a different configuration of OSGroup and TestTFM. When we generate the test list for CloudTest, we include all the zips in the test directories even the ones that have the same test name but different OSGroup/TestTFM. The idea behind this is to filter every test for the appropriate testTFM and fallback no the appropriate OSGroup.